chat-client/src/client/mynahUi.test.ts

```diff
@@ -0,0 +1,153 @@
+            followUpClicked: sinon.stub(),
+            sendFeedback: sinon.stub(),
+            linkClick: sinon.stub(),
+            sourceLinkClick: sinon.stub(),
+            infoLinkClick: sinon.stub(),
+            uiReady: sinon.stub(),
+        }
+
+        messager = new Messager(chatApi)
+        onChatPromptSpy = sinon.spy(messager, 'onChatPrompt')
+        onQuickActionSpy = sinon.spy(messager, 'onQuickActionCommand')
+
+        const tabFactory = new TabFactory({})
+        createTabStub = sinon.stub(tabFactory, 'createTab')
+        createTabStub.returns({})
+        const mynahUiResult = createMynahUi(messager, tabFactory)
+        mynahUi = mynahUiResult[0]
+        api = mynahUiResult[1]
+        getSelectedTabIdStub = sinon.stub(mynahUi, 'getSelectedTabId')
+        updateStoreSpy = sinon.spy(mynahUi, 'updateStore')
+        addChatItemSpy = sinon.spy(mynahUi, 'addChatItem')
+    })
+
+    afterEach(() => {
+        sinon.resetHistory()
+    })
+
+    it('should handle normal chat prompt', () => {
+        const tabId = 'tab-1'
+        const prompt = { prompt: 'Test prompt', escapedPrompt: 'Test prompt' }
+
+        handleChatPrompt(mynahUi, tabId, prompt, messager)
+
+        assert.notCalled(onQuickActionSpy)
+        assert.calledWith(onChatPromptSpy, { prompt, tabId })
+        assert.calledWith(addChatItemSpy, tabId, { type: ChatItemType.PROMPT, body: prompt.escapedPrompt })
+        assert.calledWith(updateStoreSpy, tabId, { loadingChat: true, promptInputDisabledState: true })
+        assert.calledWith(addChatItemSpy, tabId, { type: ChatItemType.ANSWER_STREAM })
+    })
+
+    it('should handle clear quick action', () => {
+        const tabId = 'tab-1'
+        const prompt = { prompt: 'Test prompt', escapedPrompt: 'Test prompt', command: '/clear' }
+
+        handleChatPrompt(mynahUi, tabId, prompt, messager)
+
+        assert.notCalled(onChatPromptSpy)
+        assert.calledWith(onQuickActionSpy, { quickAction: prompt.command, prompt: prompt.prompt, tabId })
+        assert.calledThrice(updateStoreSpy)
+        assert.calledWith(updateStoreSpy.firstCall, tabId, { chatItems: [] })
+        assert.calledWith(updateStoreSpy.secondCall, tabId, { loadingChat: false, promptInputDisabledState: false })
+        assert.calledWith(updateStoreSpy.thirdCall, tabId, { loadingChat: true, promptInputDisabledState: true })
+    })
+
+    it('should handle quick actions', () => {
+        const tabId = 'tab-1'
+        const prompt = { prompt: 'Test prompt', escapedPrompt: 'Test prompt', command: '/help' }
+
+        handleChatPrompt(mynahUi, tabId, prompt, messager)
+
+        assert.notCalled(onChatPromptSpy)
+        assert.calledWith(onQuickActionSpy, { quickAction: prompt.command, prompt: prompt.prompt, tabId })
+        assert.calledOnce(updateStoreSpy)
+        assert.calledWith(updateStoreSpy, tabId, { loadingChat: true, promptInputDisabledState: true })
+    })
+
+    it('should create a new tab if none exits', () => {
```

Comment:
It would be good to specify somewhere in test name (or better separate `describe`) that those tests are for `sendGenericCommand`.